### DISCUSSION TITLE
Add support for JoyconDroid

### DISCUSSION THREAD

**Discussion by maximxlss** [ndeadly/MissionControl#593] [Ideas]
It seems that on HOS 12.0.0+ JoyconDroid [requires you to spoof the device class](https://joycon.gitbook.io/joycondroid/compatibility) to work. This can be done with root, but isn't possible on MIUI and unrooted phones. It seems this project has the capabilities to make it work on any device.

**Comment by ndeadly** [comment]
It might be possible to achieve this without even involving Mission Control directly at all. I distribute exefs patches that (amongst other things) relax the gamepad device class checks introduced in 12.0.0 to include devices identifying as joysticks, mouse and keyboard. This could be extended to add additional device classes, or even patched out completely. The patches are loaded by Atmosphere independently of the Mission Control sysmodule, so could be distributed as a standalone package.

The main issue I see with this, is that it could potentially mess up the detection of bluetooth audio devices (the reason I believe the checks were added in the first place). It could also potentially blow out the number of devices which will be automatically paired to the console while on the Change Grip/Order screen (imagine using your console on public transport and it starts connecting to every nearby phone with bluetooth enabled).

I will investigate the problem and see if it's not possible to somehow make an exception for JoyconDroid devices without exposing the above issues.
